core/change.go

```diff
@@ -100,85 +101,132 @@ func (change Change) Write(writer io.Writer) error {
 	return err
 }
 
-func kindFromLabel(config Config, label string) *KindConfig {
-	for _, kindConfig := range config.Kinds {
-		if kindConfig.Label == label {
-			return &kindConfig
-		}
+// AskPrompts will ask the user prompts based on the configuration
+// updating the change as prompts are answered.
+func AskPrompts(change *Change, config Config, stdinReader io.ReadCloser) error {
+	p := Prompter{
+		change:      change,
+		config:      config,
+		stdinReader: stdinReader,
+		kind:        nil,
+		err:         nil,
 	}
 
-	panic("label not part of any kind")
+	return p.Prompt()
 }
 
-// AskPrompts will ask the user prompts based on the configuration
-// updating the change as prompts are answered.
-func AskPrompts(change *Change, config Config, stdinReader io.ReadCloser) error {
-	var (
-		err  error
-		kind *KindConfig
-	)
-
-	if len(config.Components) > 0 {
-		compPrompt := promptui.Select{
-			Label: "Component",
-			Items: config.Components,
-			Stdin: stdinReader,
-		}
+type Prompter struct {
+	change      *Change
+	config      Config
+	stdinReader io.ReadCloser
+	kind        *KindConfig
+	err         error
+}
 
-		_, change.Component, err = compPrompt.Run()
-		if err != nil {
-			return err
-		}
+func (p *Prompter) Prompt() error {
+	p.promptForComponent()
+	p.promptForKind()
+	p.parseKind()
+	p.promptForBody()
+	p.promptForUserChoices()
+
+	return p.err
+}
+
+func (p *Prompter) promptForComponent() {
+	if len(p.config.Components) == 0 {
+		return
 	}
 
-	if len(config.Kinds) > 0 {
-		kindPrompt := promptui.Select{
-			Label: "Kind",
-			Items: config.Kinds,
-			Stdin: stdinReader,
-		}
+	compPrompt := promptui.Select{
+		Label: "Component",
+		Items: p.config.Components,
+		Stdin: p.stdinReader,
+	}
 
-		_, change.Kind, err = kindPrompt.Run()
-		if err != nil {
-			return err
-		}
+	_, p.change.Component, p.err = compPrompt.Run()
+}
 
-		kind = kindFromLabel(config, change.Kind)
+func (p *Prompter) promptForKind() {
+	if p.err != nil {
+		return
 	}
 
-	if kind == nil || !kind.SkipBody {
-		bodyPrompt := config.Body.CreatePrompt(stdinReader)
-		change.Body, err = bodyPrompt.Run()
+	if len(p.config.Kinds) == 0 || len(p.change.Kind) > 0 {
+		return
+	}
+
+	kindPrompt := promptui.Select{
+		Label: "Kind",
+		Items: p.config.Kinds,
+		Stdin: p.stdinReader,
+	}
 
-		if err != nil {
-			return err
+	_, p.change.Kind, p.err = kindPrompt.Run()
+}
+
+func (p *Prompter) parseKind() {
+	if p.err != nil {
+		return
+	}
+
+	if len(p.change.Kind) == 0 {
+		return
+	}
+
+	for i := range p.config.Kinds {
+		kindConfig := &p.config.Kinds[i]
+
+		if kindConfig.Label == p.change.Kind {
+			p.kind = kindConfig
+			return
 		}
 	}
 
-	change.Custom = make(map[string]string)
+	p.err = errors.New("invalid kind")
```

Comment:
Create an `errInvalidKind` for this error so we can use it in tests as well. 
That is, instead of `Expect(...).NotToSucceed()` you should use `Expect(...).To(Equal(errInvalidKind))`

Reply:
Done that too

Reply:
Oh I saw there was a collection of these kinds of errors in `custom.go` already, I guess I'll mix that in there.

Reply:
Hmm I would keep it in this file as it is more related